### PR TITLE
Add algorithm for Casimir Effect

### PR DESCRIPTION
**Pull request by ZeroDayOwl**
### Describe your change:
I implemented an algorithm to find out the magnitudes of either Force of attraction, distance or surface area of any of the two parallel plates in vacuum between which there is an existence of Casimir effect, using the general equation for Casimir effect, provided the values of the any of the two parameters are non-zero numbers.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/casimir_effect.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# Pi and the function
+REDUCED_PLANCK_CONSTANT = 1.054571817e-34  # unit of ℏ : J * s
+
+SPEED_OF_LIGHT = 3e8  # unit of c : m * s^-1
+
+PI = 3.14
+
+
+def casimir_force(force: float, area: float, distance: float) -> dict[str, float]:
+
```
Comment (CaedenPH) [inline physics/casimir_effect.py:50]:
```suggestion
```

**Review by CaedenPH — COMMENTED** [review]
Instead of passing `0`, how about all arguments are `None`, and whichever argument isnt passed is evaluated. Otherwise, something may break if a legitimate value of a parameter is `0`

In addition, instead of returning a dictionary of the missing value, you can just return the missing value

File: physics/casimir_effect.py (hunk 1 above)
Comment (ZeroDayOwl) [reply physics/casimir_effect.py:50]: @CaedenPH, thank you for your suggestions. Technically if the legitimate value of a parameter is zero, the value of the other two parameters in the equation become `0` as well. That is why the argument which is to be evaluated by the algorithm can be `0` here. Another reason of using this is to explicitly let the user know which arguments are passed on to the function.

Moreoever, I wanted to return the data in key:value pair, so I used the dictionary to return the value.

**Comment by ZeroDayOwl** [comment]
Hey @cclauss could you kindly take a look at the PR?

File: physics/casimir_effect.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+from __future__ import annotations
+
+# Define the Reduced Planck Constant ℏ (H bar), speed of light C, value of
+# Pi and the function
+REDUCED_PLANCK_CONSTANT = 1.054571817e-34  # unit of ℏ : J * s
+
+SPEED_OF_LIGHT = 3e8  # unit of c : m * s^-1
+
+PI = 3.14
```
Comment (cclauss) [inline physics/casimir_effect.py:46]: `from math import pi`

**Review by cclauss — APPROVED** [review]
Nice!

File: physics/casimir_effect.py (hunk 1 above)
Comment (CaedenPH) [reply physics/casimir_effect.py:50]: @cclauss 
I'm willing to say that returning in a key:value is better, but what is your opinion?
To me it seems like doing
```py
def add(a, b):
    return {"sum": a+b}
```

File: physics/casimir_effect.py (hunk 2 above)
Comment (ZeroDayOwl) [reply physics/casimir_effect.py]: @cclauss Thanks for the suggestion! I used the module, and also updated the doc tests accordingly!

**Comment by ZeroDayOwl** [comment]
@cclauss Thanks for the suggestion! I used the module, and also updated the doc tests accordingly!

**Pull request merged**